### ISSUE TITLE
'consistent-this' does not accept multiple aliasses

### ISSUE THREAD

**Issue by DerZyklop** [eslint/eslint#5315]
The [documentation](http://eslint.org/docs/rules/consistent-this) says `"consistent-this": [ 2, "self",  "vm" ]` should work, but my eslint grunt task says 

> Configuration for rule "consistent-this" is invalid: Value "2,self,vm" has more items than allowed.

**Comment by platinumazure** [comment]
Looks like the docs are incorrect. The aliases need to be in their own array: `"consistent-this": [2, ["self", "vm"]]`

Can you try that out and make sure that works for you? If so, then we just have a docs bug. If that doesn't work, we have different problems :cry:

**Comment by ilyavolodin** [comment]
@platinumazure Is right, looks like documentation is incorrect. It should be an array inside an array.

**Comment by alberto** [comment]
Yes, the doc is wrong, but it is not only a doc issue. This rule was updated to accept more aliases, but that was a breaking change. Should we add back the original option to make it non breaking? @eslint/eslint-team Or should we just add it as a breaking change for 2.0.0 in the docs?

**Comment by ilyavolodin** [comment]
Hmm... I assume previously it just took a single string instead of the array? In that case, yes, looks like an oversight on our part. We should probably add back original option for backwards compatibility.

**Comment by alberto** [comment]
Yes.

**Comment by alberto** [comment]
Working on this

**Comment by mysticatea** [comment]
https://github.com/eslint/eslint/blob/master/tests/lib/rules/consistent-this.js#L51
Is this not working?

**Comment by ilyavolodin** [comment]
Hmm.. That's very strange. That should be failing schema validation:
https://github.com/eslint/eslint/blob/master/lib/rules/consistent-this.js#L132

**Comment by mysticatea** [comment]
Looks it's working.

```js
/*eslint consistent-this:[2, "a", "b"]*/
var a = this;
var b = this;
var c = this;
```

```
> eslint -v
v2.1.0
> eslint test.js --no-ignore --no-eslintrc

C:\Users\t-nagashima.AD\Documents\GitHub\eslint\test.js
  5:5  error  Unexpected alias 'c' for 'this'  consistent-this

✖ 1 problem (1 error, 0 warnings)
```

I think we should write the version options were added.

**Comment by ilyavolodin** [comment]
That looks like bug in schema validation in that case. @btmills any ideas?

**Comment by mysticatea** [comment]
I believe schema validation is correct.

```js
// rule: [2, "a", "b", "c"]
module.exports.schema = {
    "type": "array",
    "items": {
        "type": "string",
        "minLength": 1
    },
    "uniqueItems": true
};

// rule: [2, ["a", "b", "c"]]
module.exports.schema = [
    {
        "type": "array",
        "items": {
            "type": "string",
            "minLength": 1
        },
        "uniqueItems": true
    }
];
```

**Comment by mysticatea** [comment]
This rule has consistency in docs, tests, and implementation.
It's intentional and not a breaking change, I believe.

Just the additional aliases' feature has been supported in 2.0.0 and later.

@DerZyklop What is the version of eslint you are using?

**Comment by ilyavolodin** [comment]
Ahh.. you are correct, I should've looked closer. In that case, it looks like @DerZyklop might be using older version of ESLint.

**Comment by alberto** [comment]
Yes, after looking closer, it's working fine. The schema tricked me up. Sorry for causing confusion here.

It works in 2.1.0 and fails with the above message in 1.10.3.

**Comment by DerZyklop** [comment]
You are right. My package was **1.10.3**. 

I was confused because i thought `npm install --save-dev eslint` or at least `npm update --save-dev eslint` would give me the latest version. Sorry guys.

**Comment by alberto** [comment]
`npm install` and `npm update` restrict versions to what's configured in `package.json`. If you had `^1.10.3` it won't install or update to 2.0.0 or higher.
